sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java

```diff
@@ -293,60 +414,84 @@ private Mono<Void> reportMetrics(Mono<Void> checkpointMono, Checkpoint checkpoin
+                .addKeyValue(PARTITION_ID_LOG_KEY, partitionId)
+                .log(Messages.CLAIM_ERROR + ". Not BlobErrorCode.BLOB_NOT_FOUND.", error);
+            return false;
+        } else {
+            return true;
+        }
     }
 
-    private String getBlobName(String fullyQualifiedNamespace, String eventHubName, String consumerGroupName,
-        String partitionId, String typeSuffix) {
-        return fullyQualifiedNamespace + BLOB_PATH_SEPARATOR + eventHubName + BLOB_PATH_SEPARATOR + consumerGroupName
-            + typeSuffix + partitionId;
+    private static String getBlobPrefix(String fullyQualifiedNamespace, String eventHubName, String consumerGroupName,
+        String typeSuffix, boolean isLowercase) {
+
+        final String prefix = fullyQualifiedNamespace + BLOB_PATH_SEPARATOR + eventHubName + BLOB_PATH_SEPARATOR
+            + consumerGroupName + typeSuffix;
+
+        return isLowercase ? prefix.toLowerCase(Locale.ROOT) : prefix;
     }
 
-    private Mono<PartitionOwnership> convertToPartitionOwnership(BlobItem blobItem) {
-        LOGGER.atVerbose()
-            .addKeyValue(BLOB_NAME_LOG_KEY, blobItem.getName())
-            .log(Messages.FOUND_BLOB_FOR_PARTITION);
-
-        String[] names = blobItem.getName().split(BLOB_PATH_SEPARATOR);
-        if (names.length == 5) {
-            // Blob names should be of the pattern
-            // fullyqualifiednamespace/eventhub/consumergroup/ownership/<partitionId>
-            // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
-            if (CoreUtils.isNullOrEmpty(blobItem.getMetadata())) {
-                LOGGER.atWarning()
-                    .addKeyValue(BLOB_NAME_LOG_KEY, blobItem.getName())
-                    .log(Messages.NO_METADATA_AVAILABLE_FOR_BLOB);
-                return Mono.empty();
-            }
+    /**
+     * Updates a blob's metadata.  If there is an error other than 404, will propagate the error downstream.
+     * Otherwise {@code createIfNotExists} is true, will try to create the blob. Else, completes with nothing.
+     *
+     * @param client Blob client to use.
+     * @param metadata Metadata to update blob with.
+     * @param blobRequestConditions Conditions for update.
+     * @param partitionId Partition id.
+     * @param createIfNotExists True to create if blob does not exist, false otherwise.
+     *
+     * @return HttpHeaders for successful update/creation. Empty Mono if the blob does not exist and
+     *     {@code createIfNotExists} is false.  Error if the update results in an exception that is not a 404.
+     */
+    private static Mono<HttpHeaders> updateBlobMetadata(BlobAsyncClient client, Map<String, String> metadata,
+        BlobRequestConditions blobRequestConditions, String partitionId, boolean createIfNotExists) {
+
+        return client.setMetadataWithResponse(metadata, blobRequestConditions)
+            .map(response -> response.getHeaders())
```

Comment:
Could this map be put at the end and have the onErrorResume remove its map call like this?